mutex_protected.h

```diff
@@ -240,9 +246,9 @@ class mutex_protected {
 };
 
 template <typename... MutexProtected>
-auto lock(MutexProtected &...mp) {
-  std::lock(mp.mutex...);
-  return std::make_tuple(mp.adopt_lock()...);
+auto lock(MutexProtected &...mps) {
+  (mps.mutex.lock(), ...);
```

Comment:
This is a bug. If you lock in an inconsistent order (some places (a, b) and other places (b, a)) you can get deadlocks. The usual solution is to lock in a consistent order (ie sort by memory address), which `std::lock` does for us, so I'd really like to delegate to it.